MekHQ/src/mekhq/campaign/finances/Finances.java

```diff
@@ -111,37 +109,42 @@ public boolean isInDebt() {
         return getLoanBalance().isPositive();
     }
 
-    public int getFullYearsInDebt(GregorianCalendar cal) {
-        if (null == wentIntoDebt) {
+    public int getFullYearsInDebt(LocalDate date) {
+        if (wentIntoDebt == null) {
             return 0;
+        } else {
+            return Math.toIntExact(ChronoUnit.YEARS.between(wentIntoDebt, date));
         }
-        return Utilities.getDiffFullYears(wentIntoDebt, cal);
     }
 
-    public int getPartialYearsInDebt(GregorianCalendar cal) {
-        if (wentIntoDebt == null) {
-            return 0;
+    public int getPartialYearsInDebt(LocalDate date) {
+        if (wentIntoDebt != null) {
+            Period period = Period.between(wentIntoDebt, date);
+            if ((period.getMonths() > 0) || (period.getDays() > 0)) {
```

Comment:
Does this logic match the old logic, the old logic gave you `days % 365 > 0 ? 1 : 0`. Is this equivalent?

Reply:
It does. Period in single calculations will split the time between into days, months, and years. If there is no difference in a category, the value will be 0.